test/types/request_status_test.dart

```diff
@@ -0,0 +1,238 @@
+import 'package:convenience_types/errors/app_error.dart';
+import 'package:convenience_types/types/request_status.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
```

Comment:
Does it make sense to have tests, to validate the operation of data and error getters?

Something like this:
```
const String mockValue = 'teste';
RequestStatus testRequestStatus = const Succeeded(mockValue);

expect(testRequestStatus.data, mockValue);
expect(testRequestStatus.data, isA<String>());
```